cmd/server/storage/repositories/postgres/postgres.go

```diff
@@ -0,0 +1,125 @@
+package postgresdb
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+
+	"github.com/jmoiron/sqlx"
+	_ "github.com/lib/pq"
+	log "github.com/sirupsen/logrus"
+	"github.com/vivalavoka/go-exporter/cmd/server/config"
+	"github.com/vivalavoka/go-exporter/internal/metrics"
+)
+
+type PostgresDB struct {
+	config     config.Config
+	connection *sqlx.DB
+	insertStmt *sql.Stmt
+}
+
+func New(cfg config.Config) (*PostgresDB, error) {
+	conn, err := sqlx.Open("postgres", cfg.DatabaseDSN)
+	if err != nil {
+		return nil, fmt.Errorf("unable to connect to database: %v", err)
```

Comment:
Лучше использовать модификатор %w иначе мы не сможет чекать цепочку ошибок https://go.dev/play/p/qhu7aF_KY7D
```suggestion
		return nil, fmt.Errorf("unable to connect to database: %w", err)
```